src/geode/inspector/topology/internal/expected_nb_cmvs.cpp

```diff
@@ -167,40 +169,110 @@ namespace
                           nb_block_cmvs, " Block mesh vertices (should be ",
                           predicted_nb_block_cmvs, ")." ) ) );
         }
-        const auto nb_free_line_cmvs = count_cmvs(
-            unique_vertex_cmvs.line_cmvs, [&brep]( const auto& cmv ) {
-                return brep.nb_incidences( cmv.component_id.id() ) == 1
-                       && brep.nb_embedding_surfaces(
-                              brep.line( cmv.component_id.id() ) )
-                              == 0;
+        // const auto nb_free_line_cmvs = count_cmvs(
+        //     unique_vertex_cmvs.line_cmvs, [&brep]( const auto& cmv ) {
+        //         return brep.nb_incidences( cmv.component_id.id() ) == 1
+        //                && brep.nb_embedding_surfaces(
+        //                       brep.line( cmv.component_id.id() ) )
+        //                       == 0;
+        //     } );
+        // auto predicted_nb_block_cmvs =
+        //     nb_internal_surface_cmvs < nb_free_line_cmvs + 1
+        //         ? static_cast< geode::index_t >( 1 )
+        //         : nb_internal_surface_cmvs - nb_free_line_cmvs;
+        // if( nb_internal_surface_cmvs - nb_free_line_cmvs == 1 )
+        // {
+        //     predicted_nb_block_cmvs++;
+        // }
+        // if( nb_boundary_surface_cmvs > 1
+        //     && unique_vertex_cmvs.corner_cmvs.empty() )
+        // {
+        //     predicted_nb_block_cmvs += ( nb_boundary_surface_cmvs - 2 ) / 2;
+        // }
+        /// predicted_nb_block_cmvs = 1 + nb_internal_surfaces - nb_free_lines -
+        /// nb_lines_boundary_to_strictly_2_internal_surfaces
+        absl::flat_hash_set< geode::uuid > uuids_used;
+        const auto nb_internal_surfaces = count_cmvs(
+            unique_vertex_cmvs.surface_cmvs,
+            [&block_uuid, &brep, &uuids_used]( const auto& cmv ) {
+                if( !uuids_used.emplace( cmv.component_id.id() ).second )
+                {
+                    return false;
+                }
+                return brep.is_internal( brep.surface( cmv.component_id.id() ),
+                    brep.block( block_uuid ) );
             } );
-        auto predicted_nb_block_cmvs =
-            nb_internal_surface_cmvs < nb_free_line_cmvs + 1
-                ? static_cast< geode::index_t >( 1 )
-                : nb_internal_surface_cmvs - nb_free_line_cmvs;
-        if( nb_internal_surface_cmvs - nb_free_line_cmvs == 1 )
-        {
-            predicted_nb_block_cmvs++;
-        }
-        if( nb_boundary_surface_cmvs > 1
-            && unique_vertex_cmvs.corner_cmvs.empty() )
-        {
-            predicted_nb_block_cmvs += ( nb_boundary_surface_cmvs - 2 ) / 2;
-        }
+        uuids_used.clear();
+        /// Lines that are free but incident to the same surface are counted
+        /// only once
+        const auto nb_free_lines = count_cmvs( unique_vertex_cmvs.line_cmvs,
```

Comment:
I think we still should count the cmvs of those lines et those surfaces, because eg. number of lines depends on how the lines are split by corners far from studied unique vertex, without impacting the expected number of block cmv